src/libre/re_ast.h

```diff
@@ -31,13 +31,17 @@
 enum ast_expr_type {
 	AST_EXPR_EMPTY,
 	AST_EXPR_CONCAT,
+	AST_EXPR_CONCAT_N,
 	AST_EXPR_ALT,
+	AST_EXPR_ALT_N,
 	AST_EXPR_LITERAL,
 	AST_EXPR_ANY,
 	AST_EXPR_REPEATED,
 	AST_EXPR_CHAR_CLASS,
 	AST_EXPR_GROUP,
-	AST_EXPR_FLAGS
+	AST_EXPR_FLAGS,
+	AST_EXPR_ANCHOR,
+	AST_EXPR_TOMBSTONE
```

Comment:
I think the proper term for this is a "trap state", but I might be wrong.

Reply:
I think it's "trap state" in automata literature, and "tombstone" in persistent data structure literature. I'm thinking about it more as the latter, because it's flagging a location in a structure whose previous value has been explicitly removed already and could be re-used, but doesn't force a global restructuring / rebalancing -- like a deleted entry in a hash table (that can be refilled, but is a distinct state from empty when searching), or a removed entry in a persistent tree or CRDT.